thorclient/thorclient.go

```diff
@@ -0,0 +1,274 @@
+func (c *Client) InspectTxClauses(tx *tx.Transaction, senderAddr *thor.Address, opts ...Option) ([]*accounts.CallResult, error) {
+	clauses := convertToBatchCallData(tx, senderAddr)
+	return c.InspectClauses(clauses, opts...)
+}
+
+// AccountCode retrieves the account code for a given address.
+func (c *Client) AccountCode(addr *thor.Address, opts ...Option) (*accounts.GetCodeResult, error) {
+	options := applyOptions(opts)
+	return c.httpConn.GetAccountCode(addr, options.revision)
+}
+
+// AccountStorage retrieves the storage value for a given address and key.
+func (c *Client) AccountStorage(addr *thor.Address, key *thor.Bytes32, opts ...Option) (*accounts.GetStorageResult, error) {
+	options := applyOptions(opts)
+	return c.httpConn.GetAccountStorage(addr, key, options.revision)
+}
+
+// Transaction retrieves a transaction by its ID.
+func (c *Client) Transaction(id *thor.Bytes32, opts ...Option) (*transactions.Transaction, error) {
+	options := applyOptions(opts)
+	return c.httpConn.GetTransaction(id, options.revision, options.pending)
+}
+
+// RawTransaction retrieves the raw transaction data by its ID.
+func (c *Client) RawTransaction(id *thor.Bytes32, opts ...Option) (*transactions.RawTransaction, error) {
+	options := applyOptions(opts)
+	return c.httpConn.GetRawTransaction(id, options.revision, options.pending)
+}
+
+// TransactionReceipt retrieves the receipt for a transaction by its ID.
+func (c *Client) TransactionReceipt(id *thor.Bytes32, opts ...Option) (*transactions.Receipt, error) {
+	options := applyOptions(opts)
+	return c.httpConn.GetTransactionReceipt(id, options.revision)
+}
+
+// SendTransaction sends a signed transaction to the blockchain.
+func (c *Client) SendTransaction(tx *tx.Transaction) (*transactions.SendTxResult, error) {
+	rlpTx, err := rlp.EncodeToBytes(tx)
+	if err != nil {
+		return nil, fmt.Errorf("unable to encode transaction - %w", err)
+	}
+
+	return c.SendRawTransaction(rlpTx)
+}
+
+// SendRawTransaction sends a raw RLP-encoded transaction to the blockchain.
+func (c *Client) SendRawTransaction(rlpTx []byte) (*transactions.SendTxResult, error) {
+	return c.httpConn.SendTransaction(&transactions.RawTx{Raw: hexutil.Encode(rlpTx)})
+}
+
+// Block retrieves a block by its revision.
+func (c *Client) Block(revision string) (blocks *blocks.JSONCollapsedBlock, err error) {
+	return c.httpConn.GetBlock(revision)
+}
+
+// ExpandedBlock retrieves an expanded block by its revision.
+func (c *Client) ExpandedBlock(revision string) (blocks *blocks.JSONExpandedBlock, err error) {
+	return c.httpConn.GetExpandedBlock(revision)
+}
+
+// FilterEvents filters events based on the provided filter request.
+func (c *Client) FilterEvents(req *events.EventFilter) ([]events.FilteredEvent, error) {
+	return c.httpConn.FilterEvents(req)
+}
+
+// FilterTransfers filters transfers based on the provided filter request.
+func (c *Client) FilterTransfers(req *events.EventFilter) ([]*transfers.FilteredTransfer, error) {
```

Comment:
Should be `TransferFilter`

Reply:
We already have `filter - transfer` style in the code, like [here](https://github.com/vechain/thor/blob/master/api/transfers/transfers.go#L58), also the returned data structure is called `FilteredTransfer`. I would keep the naming consistent, wdyt?

Reply:
Sorry for the confusion, the parameter in the function. Should be `github.com/vechain/thor/api/transfers.TransferFilter`

Reply:
Ohh, you're right, good catch!